code/modules/mining/machine_processing.dm

```diff
@@ -404,17 +404,14 @@
 			continue
 
 		var/obj/item/stack/ore/O = A
-		if(!O.material)
+		if(!O.materials)
+			if(O.material) //legacy goonores
+				ore.addAmount(O.material, O.amount)
```

Comment:
This doesn't work anyways. Legacy goon ores don't have a proper material definition so all this does is dump a warning to `world.log`.

Reply:
It does work, it was tested with mythril